apps/wallet-dashboard/app/(protected)/vesting/page.tsx

```diff
@@ -264,152 +256,174 @@ function VestingDashboardPage(): JSX.Element {
+                                <div className="flex w-full flex-col items-center justify-center space-y-4 pt-4">
+                                    {system &&
+                                        timelockedStakedObjectsGrouped?.map(
+                                            (timelockedStakedObject) => {
+                                                return (
+                                                    <StakedTimelockObject
+                                                        key={
+                                                            timelockedStakedObject.validatorAddress +
+                                                            timelockedStakedObject.stakeRequestEpoch +
+                                                            timelockedStakedObject.label
+                                                        }
+                                                        getValidatorByAddress={
+                                                            getValidatorByAddress
+                                                        }
+                                                        timelockedStakedObject={
+                                                            timelockedStakedObject
+                                                        }
+                                                        handleUnstake={handleUnstake}
+                                                        currentEpoch={Number(system.epoch)}
+                                                    />
+                                                );
+                                            },
+                                        )}
+                                </div>
+                            </div>
+                        </Panel>
+                    </div>
+                ) : null}
+
+                {isDialogStakeOpen && (
+                    <StakeDialog
+                        isTimelockedStaking
+                        stakedDetails={selectedStake}
+                        onSuccess={handleOnSuccess}
+                        handleClose={handleCloseStakeDialog}
+                        view={stakeDialogView}
+                        setView={setStakeDialogView}
+                        selectedValidator={selectedValidator}
+                        setSelectedValidator={setSelectedValidator}
+                        maxStakableTimelockedAmount={BigInt(vestingSchedule.availableStaking)}
+                        onUnstakeClick={() => setUnstakeDialogOpen(true)}
+                    />
+                )}
+
+                {isUnstakeDialogOpen && timelockedObjectsToUnstake && (
+                    <UnstakeDialog
+                        groupedTimelockedObjects={timelockedObjectsToUnstake}
+                        view={unstakeDialogView}
+                        handleClose={() => setUnstakeDialogOpen(false)}
+                        onSuccess={() => setUnstakeDialogOpen(false)}
```

Comment:
we should refetch the data so the unstaked cards dont remain in the list after unstaking